contracts/HATVaultsRegistry.sol

```diff
@@ -0,0 +1,396 @@
+    using SafeERC20 for IERC20;
+
+    // Used in {swapAndSend} to avoid a "stack too deep" error
+    struct SwapData {
+        uint256 totalHackersHatReward;
+        uint256 amount;
+        uint256 amountUnused;
+        uint256 hatsReceived;
+        uint256 totalHackerReward;
+    }
+
+    uint256 public constant HUNDRED_PERCENT = 10000;
+
+    address public immutable hatVaultImplementation;
+    address[] public hatVaults;
+    mapping(address => bool) public isVaultVisible;
+
+    // PARAMETERS FOR ALL VAULTS
+    IHATVaultsRegistry.GeneralParameters public generalParameters;
+    ITokenLockFactory public immutable tokenLockFactory;
+    // feeSetter sets the withdrawal fee
+    address public feeSetter;
+
+    // the token into which a part of the the bounty will be swapped into
+    IERC20 public immutable HAT;
+
+    // asset => hacker address => amount
+    mapping(address => mapping(address => uint256)) public hackersHatReward;
+    // asset => amount
+    mapping(address => uint256) public governanceHatReward;
+
+    IHATVaultsRegistry.HATBountySplit public defaultHATBountySplit;
+
+    address public defaultArbitrator;
+    uint256 public defaultChallengePeriod;
+    uint256 public defaultChallengeTimeOutPeriod;
+
+    bool public isEmergencyPaused;
+
+    /**
+    * @notice initialize -
+    * @param _hatVaultImplementation The hat vault implementation address.
+    * @param _hatGovernance The governance address.
+    * @param _HAT the HAT token address
+    * @param _defaultHATBountySplit The default way to split the hat bounty betweeen
+    * the hacker and the governance
+    *   Each entry is a number between 0 and `HUNDRED_PERCENT`.
+    *   Total splits should be less than `HUNDRED_PERCENT`.
+    * @param _tokenLockFactory Address of the token lock factory to be used
+    * to create a vesting contract for the approved claim reporter.
+    */
+    constructor(
+        address _hatVaultImplementation,
+        address _hatGovernance,
+        address _HAT,
+        IHATVaultsRegistry.HATBountySplit memory _defaultHATBountySplit,
+        ITokenLockFactory _tokenLockFactory
+    ) {
+        _transferOwnership(_hatGovernance);
+        hatVaultImplementation = _hatVaultImplementation;
+        HAT = IERC20(_HAT);
+
+        validateHATSplit(_defaultHATBountySplit);
+        defaultHATBountySplit = _defaultHATBountySplit;
+        tokenLockFactory = _tokenLockFactory;
+        generalParameters = IHATVaultsRegistry.GeneralParameters({
+            hatVestingDuration: 90 days,
+            hatVestingPeriods: 90,
+            withdrawPeriod: 11 hours,
+            safetyPeriod: 1 hours,
+            setMaxBountyDelay: 2 days,
+            withdrawRequestEnablePeriod: 7 days,
+            withdrawRequestPendingPeriod: 7 days,
+            claimFee: 0
+        });
+
+        defaultArbitrator = _hatGovernance;
+        defaultChallengePeriod = 3 days;
+        defaultChallengeTimeOutPeriod = 5 weeks;
```

Comment:
shouldn't the feeSetter be set here to _hatGovernance ?

Reply:
it does not really matter i think

Reply:
then it defaults to address(0) and you have to call `setFeeSetter`